packages/psss-server/src/routes/reports/SubmitCountryWeeklyReportRoute.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { RespondingError } from '@tupaia/utils';
+import { Route } from '../Route';
+
+const SURVEY_CODE = 'PSSS_WNR';
+
+type WeeklyReportAnswers = {
+  PSSS_Sites: number;
+  PSSS_Sites_Reported: number;
+  PSSS_AFR_Cases: number;
+  PSSS_DIA_Cases: number;
+  PSSS_ILI_Cases: number;
+  PSSS_PF_Cases: number;
+  PSSS_DLI_Cases: number;
+};
+
+export class SubmitCountryWeeklyReportRoute extends Route {
+  async buildResponse() {
+    const { week } = this.req.query;
+    const { organisationUnitCode } = this.req.params;
+    const answers = mapReqBodyToAnswers(this.req.body);
+
+    const existingSurveyResponse = await this.meditrakConnection?.findSurveyResponse(
+      SURVEY_CODE,
+      organisationUnitCode,
+      week,
+    );
+
+    if (existingSurveyResponse) {
+      return this.meditrakConnection?.updateSurveyResponse(existingSurveyResponse, answers);
+    }
+
+    return this.meditrakConnection?.createSurveyResponse(
+      SURVEY_CODE,
+      organisationUnitCode,
+      week,
+      answers,
+    );
```

Comment:
Even most of this could be shared across the two handlers...a parent/child class structure could work well here?